src/main/scala/org/broadinstitute/dsde/vault/common/directives/VersioningDirectives.scala

```diff
@@ -1,17 +1,26 @@
 package org.broadinstitute.dsde.vault.common.directives
 
-import shapeless.{HList, ::, HNil}
-import spray.routing.{PathMatcher, Directive}
+import shapeless.{::, HList, HNil}
 import spray.routing.directives.BasicDirectives._
 import spray.routing.directives.PathDirectives._
+import spray.routing.{Directive, PathMatcher}
 
 object VersioningDirectives {
-  val optionalVersion: Directive[Option[Int] :: HNil] =
-    pathPrefix("v" ~ IntNumber).map(Option(_)) | provide(None)
+  val optionalVersion: Directive[Option[Int] :: HNil] = pathPrefix("v" ~ IntNumber).map(Option(_)) | provide(None)
```

Comment:
I would like a brief discussion of `path` vs. `pathPrefix` and `version` vs. `optionalVersion`, as well as why some of these have overloads for `PathMatcher` but others do not.